web/packages/teleterm/src/ui/AppUpdater/AutoUpdatesManagement.tsx

```diff
@@ -0,0 +1,333 @@
+    if (status.reason === 'no-compatible-version') {
+      source = status;
+    }
+  } else {
+    if (
+      status.source.kind === 'most-compatible' ||
+      status.source.kind === 'managing-cluster'
+    ) {
+      source = status.source;
+    }
+  }
+
+  if (!source) {
+    return [];
+  }
+
+  const candidateClusters = source.clusters
+    .filter(c => c.toolsAutoUpdate)
+    .map(c => {
+      const otherNames = c.otherCompatibleClusters.map(c => getClusterName(c));
+      const compatibility =
+        otherNames.length === 0
+          ? 'only compatible with this cluster'
+          : `also compatible with ${otherNames.join(', ')}`;
+
+      return {
+        label: getClusterName(c.clusterUri),
+        helperText: `${c.toolsVersion} client, ${compatibility}.`,
+        value: c.clusterUri,
+      };
+    });
+
+  const nonCandidateClusters = source.clusters
+    .filter(c => !c.toolsAutoUpdate)
+    .map(c => {
+      return {
+        disabled: true,
+        label: getClusterName(c.clusterUri),
+        helperText: `${c.toolsVersion} client, automatic client tools updates disabled on this cluster.`,
+        value: c.clusterUri,
+      };
+    });
+
+  const unreachableClusters = source.unreachableClusters.map(cluster => ({
+    disabled: true,
+    label: getClusterName(cluster.clusterUri),
+    helperText: `⚠︎ Unreachable cluster: ${cluster.errorMessage}`,
+    value: cluster.clusterUri,
+  }));
+
+  return [
+    ...candidateClusters,
+    ...nonCandidateClusters,
+    ...unreachableClusters,
+  ];
+}
+
+function makeContentForEnabledAutoUpdates(
+  status: AutoUpdatesEnabled,
+  getClusterName: (clusterUri: RootClusterUri) => string
+): {
+  description: string;
+  kind: 'neutral' | 'warning';
+} {
+  switch (status.source.kind) {
+    case 'env-var':
+      return {
+        kind: 'neutral',
+        description: `The app is set to stay on version ${status.version} by your device settings.`,
+      };
+    case 'managing-cluster':
+      return {
+        kind: 'neutral',
+        description: `Updates are managed by the ${getClusterName(status.source.clusterUri)} cluster, which requires client version ${status.version}.`,
+      };
+    case 'most-compatible':
+      const managingClusters = status.source.clusters
+        .filter(c => c.toolsAutoUpdate && c.toolsVersion === status.version)
+        .map(c => getClusterName(c.clusterUri));
+      const { skippedManagingClusterUri } = status.source;
+      if (skippedManagingClusterUri) {
+        const skippedManagingClusterText = skippedManagingClusterUri
+          ? `The chosen cluster ${getClusterName(skippedManagingClusterUri)} could not be used to manage updates`
+          : '';
+        return managingClusters.length === 1
+          ? {
+              kind: 'warning',
+              description: `${skippedManagingClusterText}. Updates will be managed by the ${managingClusters.at(0)} cluster, which requires client version ${status.version}.`,
```

Comment:
Also, with `most-compatible` and `skippedManagingClusterUri` ([story](https://localhost:9002/?path=/story/teleterm-appupdater--details&args=clusterBar:Unreachable;clusterBarSetToManageUpdates:!true)), the story says `clusterBarSetToManageUpdates`, but the component has "Use the most compatible version from your clusters" selected. This also hides the exact error behind bar being inaccessible.

What if "Use the most compatible version from your clusters" was another radio option?

Reply:
As discussed, we won't be hiding the options when "Use the most compatible version from your clusters" is selected. We will continue using a checkbox, as there wasn't a good reason to switch to radio.